proto/src/determined/api/v1/trial.proto

```diff
@@ -520,6 +520,30 @@ message AllocationRendezvousInfoResponse {
   determined.trial.v1.RendezvousInfo rendezvous_info = 1;
 }
 
+// Gets some allocation.
+message GetAllocationRequest {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "allocation_id" ] }
+  };
+  // The id of the allocation.
+  string allocation_id = 1;
+}
+// Response to GetAllocationRequest
+message GetAllocationResponse {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "allocation_id", "accelerator_data" ] }
+  };
+  // The id of the allocation.
+  string allocation_id = 1;
+  // The state of the allocation.
+  string state = 2;
+  // The number of slotss associated with the allocation.
+  int32 slots = 3;
+  // The exit reason for the allocation.
+  string exit_reason = 4;
```

Comment:
Can we make exit_reason and status_code optional with 

```optional exit_reason = 4```

Reply:
Done!